### PR TITLE
Add cucumber test for menu & create new 'Others' category auto.

### PR DESCRIPTION
**Pull request by namirinz**
- Just add some new cucumber test for menus.
- Auto generate category 'Others' If it's not present and user wants to create a menu without a category.

### CODE REVIEW

**Comment by lazykern** [comment]
what the fuck

**Comment by namirinz** [comment]
It's pass = It's good

**Comment by namirinz** [comment]
It passed = It's work

File: src/menus/menus.service.ts [hunk 1]
```diff
@@ -106,6 +106,17 @@ export class MenusService {
...
+        menus: {
+          $filter: {
+            input: '$menus',
+            as: 'menu',
+            cond: { $ne: ['$$menu', null] },
+          },
+        },
+      },
+    },
     {
```
Comment (namirinz) [inline src/menus/menus.service.ts:120]: This line is use to remove the null menu inside the array menu when querying with status

File: src/menus/menus.service.ts [hunk 2]
```diff
@@ -106,6 +106,17 @@ export class MenusService {
...
+      $addFields: {
+        menus: {
+          $filter: {
+            input: '$menus',
+            as: 'menu',
+            cond: { $ne: ['$$menu', null] },
+          },
+        },
+      },
+    },
```
Comment (namirinz) [inline src/menus/menus.service.ts:119]: This line is use to remove the null menu inside the array menu when querying with status

**Review by lazykern — APPROVED** [review]

**Review by ishiasaka — APPROVED** [review]
Sugoi

**Pull request merged**